### ISSUE TITLE
Error pushing to Google Container Registry: "invalid character 'I' looking for beginning of value"

### ISSUE THREAD

**Issue by paislee** [Shippable/support#1882]
I've configured a Docker build to push to Google Container Registry. Service account set up correctly. The screenshot below shows how the registry is recognized and the built image starts to be pushed.

<img width="1102" alt="screen shot 2015-09-17 at 12 02 43 pm" src="https://cloud.githubusercontent.com/assets/831742/9943103/3ca986a2-5d34-11e5-8605-74ff9f648fb3.png">

Eventually it fails with the following, and the image is never completely pushed.

<img width="1099" alt="screen shot 2015-09-17 at 12 03 00 pm" src="https://cloud.githubusercontent.com/assets/831742/9943105/3f2a6392-5d34-11e5-9f36-a8b86e1ec745.png">

I am able to build and push the project locally.

```
$ docker version
Client:
 Version:      1.8.1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   d12ea79
 Built:        Thu Aug 13 02:49:29 UTC 2015
 OS/Arch:      darwin/amd64
```

**Comment by jawaid** [comment]
@paislee  Please validate your project-Id

https://cloud.google.com/container-registry/

**Comment by paislee** [comment]
Thanks @jawaid. I have double checked that the project ID is correct in the registry endpoint. As noted, I'm successfully pushing this image to the registry outside of Shippable. Furthermore, from the Shippable logs, you can see that the entire image is actually pushed, and the failure occurs while uploading metadata (canceling the push, I'm suspecting). Shippable still sends me a success notification.

I found something relevant here: http://scottsmerchek.com/2015/07/24/pushing-to-google-container-registry-from-circleci/

> Next, we use our script from above to authenticate with gcloud, then use the gcloud docker push almost like normal. I had to add > /dev/null to ignore the output as it was causing the following error: FATA[0061] HTTP code 500 while uploading metadata: "invalid character 'I' looking for beginning of value"

**Comment by jawaid** [comment]
@paislee  We will look into this and get back to you.

**Comment by paislee** [comment]
@jawaid thanks! standing by to help

**Comment by ttrahan** [comment]
@paislee Your issue appears similar to ones discussed here:
https://github.com/docker/docker-registry/issues/941

It looks like others have experienced a similar issue when one of two situations occur:
* A previous image push was interrupted (manual deletion of suspect image in the repository is needed)
* The repo is out of space

Let us know if either of these resolve this for you.

**Comment by paislee** [comment]
HI @ttrahan. My issue is definitely not your second bullet, because I can continue to push to the same registry from outside of Shippable.

I'm beginning to think my problem is your first point, because Shippable was able to successfully push when I changed to an EU gcr endpoint. I'm guessing there are some intermediate layers stuck in a bad state from a previous error. Unfortunately, the intermediate layers don't show up in the gcr UI, so I can't delete them. I'll continue looking for solutions.

**Comment by ttrahan** [comment]
Hi @paislee : let's try to address the first bullet above. First, confirm that when you re-run the build in Shippable, that it consistently fails at the same image layer, which looks to be 'a772f5b8cd2a'. 

If so, in the Google developers console, you can view/access the individual container layers via the storage browser, which may help for this.  In the left-nav, drill down to Storage->Cloud Storage->Browser->buckets->artifacts...->containers->images.  This should bring you to folders for the individual container layers.

For example, here's an image layer created when I pushed to GCR from Shippable:
![image](https://cloud.githubusercontent.com/assets/4467650/9971081/1ee4c55c-5e0e-11e5-8822-0901ae3015e9.png)

I can find this image layer in the Storage Browser (see first folder below):
![image](https://cloud.githubusercontent.com/assets/4467650/9971059/0037fb24-5e0e-11e5-9e05-8cadd984efff.png)


Try locating your image layer starting with 'a772f5' and delete the folder, then try re-running the Shippable build. Please let us know how this goes.

**Comment by paislee** [comment]
Thanks @ttrahan. Thanks for pointing me to the Storage Browser. For simplicity's sake, I simply deleted the entire bucket. When I re-built in Shippable, there were no errors noted and the image appeared to push successfully; the build was marked successful!

However, __the image never appeared in the Google Container Registry browser__, and the tag Shippable said it pushed cannot be found. The subsequent build gets stuck in the following loop:

![screen shot 2015-09-18 at 3 29 40 pm](https://cloud.githubusercontent.com/assets/831742/9972639/65e94f6a-5e1b-11e5-868a-5063a173e9e6.png)

Should I open this as a new issue?

**Comment by paislee** [comment]
Actually @ttrahan, it appears that the build is not pushed correctly. Shippable notifies me of success before the image is done uploading. I tried it again after deleting all the image layers in GCR. Shippable notifies me of success, but the build log continues to grow as layers are pushed. The push step ends up failing with the same error reported earlier. This explains the "Image push is already in progress" from above.

The screenshot below shows how the build is successful, but the individual push step ends up failing many minutes after notification:

<img width="1183" alt="screen shot 2015-09-18 at 6 52 11 pm" src="https://cloud.githubusercontent.com/assets/831742/9974099/ba3b4ee0-5e36-11e5-9292-7bfdc016acb7.png">

Also, the same failure originally posted (this time on a clean registry).

<img width="1125" alt="screen shot 2015-09-18 at 6 52 24 pm" src="https://cloud.githubusercontent.com/assets/831742/9974102/c39ac6be-5e36-11e5-99ed-6cf74f7a64be.png">

Interestingly enough, it is the same layer hash from before.

**Comment by ttrahan** [comment]
@paislee : one of our engineers will be looking into this today.  updates to come.

**Comment by paislee** [comment]
@ttrahan thanks!

**Comment by ttrahan** [comment]
@paislee : we think that the issue is related to pushing large images.  I believe you're using minv2, which we don't recommend if you're pushing images.  Try changing from using minv2 to one of the smaller base images we have (see https://hub.docker.com/u/shippableimages/) as your 'FROM' image in your Dockerfile.

**Comment by paislee** [comment]
@ttrahan I was able to complete a successful push after realizing (due to your last comment) that Shippable was using minv2 as a base image. The docs are a bit confusing as to what happens if you specify a "push to" location without enabling the "Docker build" toggle. I incorrectly thought my project Dockerfile was being used. Once I enabled a POST-CI Docker build, my own image was built and pushed ok—assuming because it was smaller.

Incidentally, my problem is fixed, but I suppose this still stands as a bug. Thanks.
